component/identity/src/main/java/org/exoplatform/services/organization/idm/GroupDAOImpl.java

```diff
@@ -573,7 +577,7 @@ public Collection<Group> findGroupsOfUserByKeyword(String user, String keyword,
       List<Group> exoGroups = new LinkedList<Group>();
       for (org.picketlink.idm.api.Group group : allGroups) {
         try {
-          if (groupType.isEmpty() || group.getGroupType().equals(groupType)) {
+          if (groupType.isEmpty() || group.getGroupType().equals(groupType) || !group.getGroupType().equals(SPACE_GROUP_TYPE) && !group.getGroupType().equals(ROOT_GROUP_TYPE)) {
```

Comment:
I dont understand why we filter to not get groups with type SPACE_GROUP_TYPE and groups with type ROOT_GROUP_TYPE
As this function should return groups of users, if user is present in theses groups we must return it. We cannot remove arbitrary groups when someone request groups of users.

Reply:
Instead of hardcoding some grouptype to exclude, should we add in the rest service a parameter named excludedGroupTypes ?

Reply:
@IlhemEssaadi you should not filter the groups using the hardcoded type for spaces and should only return the groups with the type given in the method param: groupType.
 @rdenarie we need to filter with ROOT_GROUP_TYPE because we don't need to get root groups in the suggestor as for example (/PLatform, ...)